src/core/unit.c

```diff
@@ -3369,6 +3369,9 @@ int unit_patch_contexts(Unit *u) {
                 if (ec->private_devices)
                         ec->capability_bounding_set &= ~(UINT64_C(1) << CAP_MKNOD);
 
+                if (ec->protect_kernel_modules)
+                        ec->capability_bounding_set &= ~((UINT64_C(1) << CAP_SYS_MODULE) | (UINT64_C(1) << CAP_NET_ADMIN));
```

Comment:
so, I think we should not take CAP_NET_ADMIN away here, but only CAP_SYS_MODULE.

The rationale for this: if you have CAP_NET_ADMIN, then you can only load modules prefixed with `netdev-` but not arbitrary modules. And as such it really is more in the territory of module auto-loading, not of explicit loading. I think this patch should not bother window module auto-loading, as problems with that are limited to specific sets of kernel module names, that usually share a common prefix, and are always subject to configuration in the modules configuration or mapping files.

Yes, this deserves documentation, but only in the context of "this setting only blocks explicit module loading. Limited automatic module loading because of user configuration or kernel mapping tables might still happen as side effect of requested user operations, both privileged and unprivileged." or something like that.